pkg/setup/setup.go

```diff
@@ -0,0 +1,61 @@
+package setup
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path"
+)
+
+const ymlFileName = "docker-compose.yml"
+
+// Setup creates YAML file for Docker Compose or Docker Swarm and secrets directory.
+func Setup(d string) error {
+	if err := createYMLFile(d); err != nil {
+		return fmt.Errorf("creating YML file at %s: %w", d, err)
+	}
+	return nil
+}
+
+func createYMLFile(d string) error {
+	p := path.Join(d, ymlFileName)
+
+	pExists, err := fileExists(p)
+	if err != nil {
+		return fmt.Errorf("checking file existance: %w", err)
+	}
+	if pExists {
+		// File already exists, so skip this step.
+		return nil
+	}
+
+	w, err := os.Create(p)
+	if err != nil {
+		return fmt.Errorf("creating file `%s`: %w", p, err)
+	}
+	defer w.Close()
+
+	if err := writeContent(w); err != nil {
+		return fmt.Errorf("writing content to file %s: %w", p, err)
+	}
+	return nil
+}
+
+func writeContent(w io.Writer) error {
+	if _, err := w.Write([]byte("---\nfoo: bar\n")); err != nil {
+		return fmt.Errorf("writing content: %w", err)
+	}
+	return nil
+}
+
+func fileExists(p string) (bool, error) {
```

Comment:
There is a function `os.IsNotExist()` in the standard lib. It also contains an example how to check if an file exists: https://golang.org/pkg/os/#IsNotExist

Reply:
I read this but thex say:
` This function predates errors.Is. It only supports errors returned by the os package. New code should use errors.Is(err, os.ErrNotExist). `